generator/src/swam/generator/witx/TypesEmitTraverser.scala

```diff
@@ -0,0 +1,109 @@
+package swam.generator.witx
+
+import swam.witx.traverser.TypesTraverser
+import swam.witx.unresolved.{
+  AliasType,
+  ArrayType,
+  BaseWitxType,
+  BasicType,
+  EnumType,
+  Field,
+  FlagsType,
+  Handle,
+  Pointer,
+  StructType,
+  UnionType
+}
+
+import scala.collection.immutable.HashMap
+import scala.collection.mutable
+
+/**
+  * @author Javier Cabrera-Arteaga on 2020-03-21
+  */
+class TypesEmitTraverser(types: Map[String, BaseWitxType]) extends TypesTraverser[String](types) {
+
+  override val basicTypeTraverser = {
+    case (_, t: BasicType) =>
+      t match {
+        case BasicType.u8     => s"type ${t.tpeName} = Byte\n"
+        case BasicType.u16    => s"type ${t.tpeName} = Short\n"
+        case BasicType.u32    => s"type ${t.tpeName} = Int\n"
+        case BasicType.u64    => s"type ${t.tpeName} = Long\n"
+        case BasicType.s64    => s"type ${t.tpeName} = Long\n"
+        case BasicType.string => s"type ${t.tpeName} = String\n"
+        case BasicType.ptr    => s"type ${t.tpeName} = Int\n"
+      }
+  }
+
+  override val aliasTypeTraverser = {
+    case (_, t: AliasType) => s"type ${t.tpeName}= ${t.tpe.tpeName}\n"
+  }
+
+  override val enumTypeTraverser = {
+    case (_, t: EnumType) =>
+      s"object ${t.tpeName}Enum extends Enumeration { \n\t ${t.names
```

Comment:
Scala standard enums are really not good. Can you use [enumeratum](https://github.com/lloydmeta/enumeratum) instead? It is already a dependency of the core Swam module, so you have it here. You can have a look at how I did it in my WASI branch.